bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/things/AutomowerHandler.java

```diff
@@ -0,0 +1,285 @@
+    }
+
+    @Override
+    public void initialize() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            AutomowerConfiguration currentConfig = getConfigAs(AutomowerConfiguration.class);
+            final String configMowerId = currentConfig.getMowerId();
+            final Integer pollingIntervalS = currentConfig.getPollingInterval();
+
+            if (configMowerId == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "@text/conf-error-no-mower-id");
+            } else if (pollingIntervalS != null && pollingIntervalS < 1) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "@text/conf-error-invalid-polling-interval");
+            } else {
+                automowerId.set(configMowerId);
+                startAutomowerPolling(pollingIntervalS);
+            }
+
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
+        }
+    }
+
+    @Nullable
+    private AutomowerBridge getAutomowerBridge() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            ThingHandler handler = bridge.getHandler();
+            if (handler instanceof AutomowerBridgeHandler) {
+                AutomowerBridgeHandler bridgeHandler = (AutomowerBridgeHandler) handler;
+                return bridgeHandler.getAutomowerBridge();
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public void dispose() {
+        if (!automowerId.get().equals(NO_ID)) {
+            stopAutomowerPolling();
+            automowerId.set(NO_ID);
+        }
+    }
+
+    private void startAutomowerPolling(@Nullable Integer pollingIntervalS) {
+        if (automowerPollingJob == null) {
+            final long pollingIntervalToUse = pollingIntervalS == null ? DEFAULT_POLLING_INTERVAL_S : pollingIntervalS;
+            automowerPollingJob = scheduler.scheduleWithFixedDelay(new AutomowerPollingRunnable(), 1,
+                    pollingIntervalToUse, TimeUnit.SECONDS);
+        }
+    }
+
+    private void stopAutomowerPolling() {
+        if (automowerPollingJob != null) {
+            automowerPollingJob.cancel(true);
+            automowerPollingJob = null;
+        }
+    }
+
+    private boolean isValidResult(Mower mower) {
+        return mower.getAttributes() != null && mower.getAttributes().getMetadata() != null
+                && mower.getAttributes().getBattery() != null && mower.getAttributes().getSystem() != null;
+    }
+
+    private boolean isConnected(Mower mower) {
+        return mower.getAttributes() != null && mower.getAttributes().getMetadata() != null
+                && mower.getAttributes().getMetadata().isConnected();
+    }
+
+    private synchronized void updateAutomowerState() {
+        if (System.nanoTime() - lastQueryTimeMs > maxQueryFrequencyNanos) {
```

Comment:
So you already have a mechanism in the framework that makes sure when I query various values that a cached item is used so there are not multiple rest calls executed (of course this makes sense)?

So I can just do the query when I get a refreshChannels?

Reply:
Please have a look at my question. So you are basically saying that there is a mechanism in the framework that assures that refreshChannels is not called too often? I can remove my check completely?

Reply:
The expire class keeps track of the timeout and get a function as parameter. When the timeout exceeds it will call the function. So you can use it by simply calling `getValue()` it returns the value cached, when the time is exceeded. So it's not something the framework does. But it provides a class you can use to implement the timout functionallity you implemented here. Like not having to reinvent the wheel principle.